specify v4 addresses (java dual-stack requires an OS which allows v4 connections
on v6 sockets)

Index: conf/server.xml
--- conf/server.xml.orig
+++ conf/server.xml
@@ -67,6 +67,7 @@
          Define a non-SSL/TLS HTTP/1.1 Connector on port 8080
     -->
     <Connector port="8080" protocol="HTTP/1.1"
+               address="0.0.0.0"
                connectionTimeout="20000"
                redirectPort="8443"
                maxParameterCount="1000"
@@ -74,6 +75,7 @@
     <!-- A "Connector" using the shared thread pool-->
     <!--
     <Connector executor="tomcatThreadPool"
+               address="0.0.0.0"
                port="8080" protocol="HTTP/1.1"
                connectionTimeout="20000"
                redirectPort="8443"
@@ -89,6 +91,7 @@
     -->
     <!--
     <Connector port="8443" protocol="org.apache.coyote.http11.Http11NioProtocol"
+               address="0.0.0.0"
                maxThreads="150" SSLEnabled="true"
                maxParameterCount="1000"
                >
@@ -106,6 +109,7 @@
     -->
     <!--
     <Connector port="8443" protocol="org.apache.coyote.http11.Http11AprProtocol"
+               address="0.0.0.0"
                maxThreads="150" SSLEnabled="true"
                maxParameterCount="1000"
                >
@@ -122,7 +126,7 @@
     <!-- Define an AJP 1.3 Connector on port 8009 -->
     <!--
     <Connector protocol="AJP/1.3"
-               address="::1"
+               address="127.0.0.1"
                port="8009"
                redirectPort="8443"
                maxParameterCount="1000"
@@ -158,7 +162,7 @@
                resourceName="UserDatabase"/>
       </Realm>
 
-      <Host name="localhost"  appBase="webapps"
+      <Host name="localhost"  appBase="${CATALINA_BASE}/webapps"
             unpackWARs="true" autoDeploy="true">
 
         <!-- SingleSignOn valve, share authentication between web applications
